build.gradle

```diff
@@ -488,285 +488,286 @@ tasks.withType(Checkstyle) {
+                }
+
+                configurations.testRuntime.dependencies.each { dependency ->
+                    createDependencyNode(dependenciesNode, dependency.group, dependency.name, dependency.version, 'test', null)
+                }
+
+                // Repositories Block
+                def repositoriesNode = asNode().appendNode('repositories')
+                createRepoNode(repositoriesNode, 'jboss', 'JBoss', 'https://repository.jboss.org/nexus/content/groups/public/')
+                createRepoNode(repositoriesNode, 'oauth', 'Oauth', 'https://oauth.googlecode.com/svn/code/maven/')
+                createRepoNode(repositoriesNode, 'awood', 'awood.fedorapeople.org', 'https://awood.fedorapeople.org/ivy/candlepin/')
+                createRepoNode(repositoriesNode, 'barnabycourt', 'barnabycourt.fedorapeople.org', 'https://barnabycourt.fedorapeople.org/repo/candlepin/')
+            
+                // Build Block
+                def buildNode = asNode().appendNode('build')
+                def resourcesNode = buildNode.appendNode('resources')
+                def resource1 = resourcesNode.appendNode('resource')
+                resource1.appendNode('filtering', 'true')
+                resource1.appendNode('directory', 'src/main/resources')
+
+                def resource2 = resourcesNode.appendNode('resource')
+                resource2.appendNode('filtering', 'true')
+                resource2.appendNode('directory', '${project.build.directory}/generated-resources')
+
+                def testResourcesNode = buildNode.appendNode('testResources')
+                def testResource = testResourcesNode.appendNode('testResource')
+                testResource.appendNode('filtering', 'false')
+                testResource.appendNode('directory', 'src/test/resources')
+
+                def pluginsNode = buildNode.appendNode('plugins')
+                createOpenapiGeneratorPlugin(pluginsNode)
+                createMavenProcessorPlugin(pluginsNode)
+                createBuilderHelperMavenPlugin(pluginsNode)
+                createGettextMavenPlugin(pluginsNode)
+                createMavenResourcePlugin(pluginsNode)
+                createMavenCompilerPlugin(pluginsNode)
+                createMavenSurefirePlugin(pluginsNode)
+                createMavenAssemblyPlugin(pluginsNode)
+                createMavenCleanPlugin(pluginsNode)
+                createMavenHelpPlugin(pluginsNode)
+
+                // Profiles Block
+                def profilesNode = asNode().appendNode('profiles')
+                createProfileNode(profilesNode, 'build-with-jss4', '/usr/lib64/jss/jss4.jar', '4.9.1')
+                createProfileNode(profilesNode, 'build-with-jss5-plus', '/usr/lib64/jss/jss.jar', '5.0.0')
             }
-        }.writeTo("pom.xml")
+        }
     }
+  }
+}
+
+static void createDependencyNode(Node parentNode, String groupId, String artifactId, String version, String scope, String systemPath) {
```

Comment:
This task grew quite a lot. It should probably move to buildSrc but it depends on how long we will keep it around before migrating away from pom files.

Reply:
I'm not sure if there are plans to move away from the pom files and if it makes sense to move to the buildSrc directory or not. I'm curious on what @Ceiu thinks.

Reply:
As far as I know, the pom files are necessary due to the how our build system works and the various pieces within it. We've recently discussed moving to different build systems which wouldn't need the pom files, but I don't think we're very far along on that yet.

As far as compartmentalizing the task, if there's a clean way to do it such that it ends up in a logical place and is self-contained (i.e. not split across several files/modules), that might be better than throwing piles of stuff in the build.gradle file.

Reply:
I created a new class, PomToolkit, and moved over the pom xml creation logic to static utility methods that can be used!